src/docsfy/api/admin.py

```
@@ -0,0 +1,171 @@
+    await grant_project_access(name, username, project_owner=project_owner)
+    await notify_access_change(username)
+    logger.info(
+        f"[AUDIT] Admin '{request.state.username}' granted '{username}' access to '{name}' (owner: '{project_owner}')"
+    )
+    return {"granted": name, "username": username, "owner": project_owner}
+
+
+@router.delete("/projects/{name}/access/{username}")
+async def revoke_access(request: Request, name: str, username: str) -> dict[str, str]:
+    _require_admin(request)
+    logger.debug(f"Access revoke requested: project='{name}', username='{username}'")
```

Ruff (0.15.6)

[warning] Logging statement uses f-string

(G004)